Nimda/Core/PluginContainer.php

```diff
@@ -129,16 +129,41 @@ public function onMessage(Message $message)
             return;
         }
 
-        $plainText = trim(Str::after($plainText, $plugin->keys()->first()));
+        $arguments = $this->parseArguments($plainText, $plugin->keys()->first());
+
+        if($arguments === false) {
+            return;
+        }
 
         $plugin = $plugin->first();
-        $plugin->trigger($message, $plainText);
+        $plugin->trigger($message, $arguments);
     }
 
+    /**
+     * @param $text
+     * @return Collection
+     */
     private function findPluginByCommand($text)
     {
         return $this->commands->filter(function ($plugin) use ($text){
-            return Str::startsWith($text, array_keys($plugin)[0]);
+            $commandShard = explode(' ', array_keys($plugin)[0])[0];
+            return Str::startsWith($text, $commandShard);
         })->collapse();
     }
+
+    private function parseArguments($message, $pattern)
+    {
+        $commandRegex = '/\{((?:(?!\d+,?\d+?)\w)+?)\}/';
+        $pattern = str_replace('/', '\/', $pattern);
+        $regex = '/^'.preg_replace($commandRegex, '(?<$1>.*)', $pattern).' ?/miu';
+
+        $regexMatched = (bool)preg_match($regex, $message, $matches);
```

Comment:
Can you explain what these regex lines do? looks like a lot.

Reply:
First of all, it replaces all instances of '/' to '\/' and matches it via the pattern for valid Regex. It then creates another Regex argument based on what command has been set. I.e if the command was !purge 5, in this instance, the preg_replace would set `(?<$1>.*)` to `(?<5>.*)` and that would be parsed and set as part of an array with the name "amount" as that's whats specified inside of the plugin itself.

`$regexMatched = (bool)preg_match($regex, $message, $matches);` returns a bool based on whether the Regex match is true or false.